pyccel/parser/syntactic.py

```diff
@@ -229,24 +194,41 @@ def _visit(self, stmt):
         cls = type(stmt)
         syntax_method = '_visit_' + cls.__name__
         if hasattr(self, syntax_method):
-            return getattr(self, syntax_method)(stmt)
+            self._scope.append(stmt)
+            result = getattr(self, syntax_method)(stmt)
+            self._scope.pop()
+            return result
```

Comment:
Why is it necessary to append `stmt` to `self._scope` in order to visit `stmt`?

Reply:
Sometimes it is useful to know the parent or grandparent of the object being visited. This information is not available from the python ast. This was the easiest way to collect it

Reply:
I am a little surprised that we need the parent or the grandparent during the syntactic stage. Could you give me an example?

Reply:
Eg  a string in an expression is a comment block but a string in an assignment is a string. I believe there's one other example but I won't be able to check until Thursday

Reply:
I see, thanks!